### ISSUE TITLE
analyse_freenrg.py is broken

### ISSUE THREAD

**Issue by mark-cresset** [OpenBioSim/sire#6]
The "analyse_freenrg.py" script is currently broken. Around line 520, it tries to get the simulation temperature by parsing the input .dat file. It's assuming that it can split the line on whitespace and get the temperature from field 3 and the unit from field 4. However, the simfile.dat file now reads

#Generating temperature is  298°K

instead of

#Generating temperature is  298 K

so the parsing fails.

In addition, the code later in the script doesn't work as k_boltz now seems to have a bunch of units attached which causes an exception when it is applied to a bunch of matrix slices - I don't understand the units code well enough to work out how to fix this.

**To Reproduce**
Unpack the attached files. Run

sire_python --ignore-ipython analyse_freenrg.py mbar -i simfile_*.dat" -o freenrg-MBAR.dat --temperature 298.0 --percent 95 --overlap

This is with Sire 2023.1.1, Linux

**Comment by lohedges** [comment]
Hi @mark-cresset. Thanks for catching this. As a quick workaround for the `k_boltz` issue you can replace it with `k_bolt.value()`, which will return the value in the given unit.

**Comment by chryswoods** [comment]
Thanks - I'll take a look at the code and will sort out a fix. Thanks for supplying the simfiles. I didn't realise that the mbar part of analyse_freenrg was parsing the raw output files. We'd used s3 files for other tools to make them more robust to formatting changes.

**Comment by mark-cresset** [comment]
Thanks Chris and Lester!

**Comment by chryswoods** [comment]
I've fixed `analyse_freenrg` and confirm that it now works for the simfiles you sent. This is backwards-compatible, so should still be able to parse files produced using older versions of sire. I've submitted the pull request (above) to `devel`. Here's a [link to the changed file](https://github.com/OpenBioSim/sire/blob/fix_analyse_freenrg/wrapper/python/scripts/analyse_freenrg.py).

**Comment by chryswoods** [comment]
There were some residual bugs that needed fixing related to unicode parsing on Windows. I've fixed these, and have added an analyse_freenrg unit test to catch regressions in the future. The fix has been merged into `devel` (so can be used from the latest development binaries on conda-forge) and is now part of `2023.1.2` which is being built now. I think this should be available to download by this afternoon :-)
